lotman/lotman.go

```diff
@@ -0,0 +1,655 @@
+		MaxNumObjects   LotValueMapInt   `json:"max_num_objects"`
+		CreationTime    LotValueMapInt   `json:"creation_time"`
+		ExpirationTime  LotValueMapInt   `json:"expiration_time"`
+		DeletionTime    LotValueMapInt   `json:"deletion_time"`
+	}
+
+	UsageMapFloat struct {
+		SelfContrib     float64 `json:"self_contrib,omitempty"`
+		ChildrenContrib float64 `json:"children_contrib,omitempty"`
+		Total           float64 `json:"total"`
+	}
+
+	UsageMapInt struct {
+		SelfContrib     Int64FromFloat `json:"self_contrib,omitempty"`
+		ChildrenContrib Int64FromFloat `json:"children_contrib,omitempty"`
+		Total           Int64FromFloat `json:"total"`
+	}
+
+	LotUsage struct {
+		GBBeingWritten      UsageMapFloat `json:"GB_being_written,omitempty"`
+		ObjectsBeingWritten UsageMapInt   `json:"objects_being_written,omitempty"`
+		DedicatedGB         UsageMapFloat `json:"dedicated_GB,omitempty"`
+		OpportunisticGB     UsageMapFloat `json:"opportunistic_GB,omitempty"`
+		NumObjects          UsageMapInt   `json:"num_objects,omitempty"`
+		TotalGB             UsageMapFloat `json:"total_GB,omitempty"`
+	}
+
+	Lot struct {
+		LotName string `json:"lot_name" mapstructure:"LotName"`
+		Owner   string `json:"owner,omitempty" mapstructure:"Owner"`
+		// We don't expose Owners via map structure because that's not something we can configure. It's a derived value
+		Owners  []string `json:"owners,omitempty"`
+		Parents []string `json:"parents" mapstructure:"Parents"`
+		// While we _could_ expose Children, that complicates things so for now we keep it hidden from the config
+		Children *[]string  `json:"children,omitempty"`
+		Paths    []LotPaths `json:"paths,omitempty" mapstructure:"Paths"`
+		MPA      *MPA       `json:"management_policy_attrs,omitempty" mapstructure:"ManagementPolicyAttrs"`
+		// Again, these are derived
+		RestrictiveMPA *RestrictiveMPA `json:"restrictive_management_policy_attrs,omitempty"`
+		Usage          *LotUsage       `json:"usage,omitempty"`
+	}
+
+	ParentUpdate struct {
+		Current string `json:"current"`
+		New     string `json:"new"`
+	}
+
+	PathUpdate struct {
+		Current   string `json:"current"`
+		New       string `json:"new"`
+		Recursive bool   `json:"recursive"`
+	}
+
+	LotUpdate struct {
+		LotName string          `json:"lot_name"`
+		Owner   *string         `json:"owner,omitempty"`
+		Parents *[]ParentUpdate `json:"parents,omitempty"`
+		Paths   *[]PathUpdate   `json:"paths,omitempty"`
+		MPA     *MPA            `json:"management_policy_attrs,omitempty"`
+	}
+)
+
+// Lotman has a tendency to return an int as 123.0 instead of 123. This struct is used to unmarshal
+// those values into an int64
+func (i *Int64FromFloat) UnmarshalJSON(b []byte) error {
+	var f float64
+	if err := json.Unmarshal(b, &f); err != nil {
+		return err
+	}
+	i.Value = int64(f)
+	return nil
+}
+
+func (i Int64FromFloat) MarshalJSON() ([]byte, error) {
+	return json.Marshal(i.Value)
+}
+
+func cArrToGoArr(cArr *unsafe.Pointer) []string {
+	ptr := uintptr(*cArr)
```

Comment:
I assume it's converting a C Array to a Go Slice. Can you add a simple header explaining what this is doing?